cypress/support/customCommands/basicOperations/mailpit.js

```diff
@@ -0,0 +1,64 @@
+const mhApiUrl = path => {
+  const envValue = Cypress.env("MAILPITURL");
+  const basePath = envValue ? envValue : Cypress.config("MAILPITURL");
+  return `${basePath}/api${path}`;
+};
+
+/**
+ * Mail Collection
+ */
+
+Cypress.Commands.add("mpGetAllMails", (fromLast = 60000) =>
+  cy
+    .request({
+      method: "GET",
+      url: mhApiUrl("/v1/messages?limit=9999"),
+    })
+    .then(response => {
+      // by default get mails received in last 60000ms
+      response.body.messages = response.body.messages.filter(message => {
+        const fiveMinutesAgo = Date.now() - fromLast;
+        const mailCreated = new Date(message.Created);
+        return mailCreated.getTime() > fiveMinutesAgo;
+      });
+      if (typeof response.body === "string") {
+        return JSON.parse(response.body);
+      } else {
+        return response.body;
+      }
+    }),
+);
+
+Cypress.Commands.add("mpLatest", { prevSubject: true }, mails =>
+  Array.isArray(mails) && mails.length > 0 ? mails[0] : mails,
+);
+
+Cypress.Commands.add("mpGetMailsBySubject", subject => {
```

Comment:
Have you considered making these command chaninable (or thenable)? that instead of calling `cy.mpGetAllMails()` inside here, we chain chain them like:
```js
const lastMessageToTomaszAboutCypress = cy.mpGetAllMails.then(messages => {
 return cy.mpGetMailsByRecipient(messages,"Tomasz");
}).then(messages => {
  return cy.mpGetMailsBySubject(messages, "Cypress")l
}).then(messages => {
  return cy.mpLatest(messages);
});